lib/collection/src/tests/points_dedup.rs

```diff
@@ -284,6 +287,87 @@ async fn test_search_dedup() {
     }
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn test_query_dedup() {
+    let collection = fixture().await;
+
+    let hw_acc = HwMeasurementAcc::new();
+    let points = collection
+        .query(
+            ShardQueryRequest {
+                query: Some(ScoringQuery::Vector(QueryEnum::Nearest(
+                    NamedQuery::default_dense(vec![0.1, 0.2, 0.3, 0.4]),
+                ))),
+                prefetches: vec![],
+                filter: None,
+                params: Some(SearchParams {
+                    exact: true,
+                    ..Default::default()
+                }),
+                limit: 100,
+                offset: 0,
+                with_payload: WithPayloadInterface::Bool(false),
+                with_vector: WithVector::Bool(false),
+                score_threshold: None,
+            },
+            None,
+            ShardSelectorInternal::All,
+            None,
+            hw_acc,
+        )
+        .await
+        .expect("failed to query");
+    assert!(!points.is_empty(), "expected some points");
+
+    let mut seen = HashSet::new();
+    for point_id in points.iter().map(|point| point.id) {
+        assert!(
+            seen.insert(point_id),
+            "got point id {point_id} more than once, they should be deduplicated",
+        );
+    }
+}
```

Comment:
This new test failed before all the other changes in this PR.

We expect the point deduplication behavior of the search and query API to be the same, this test showed that was not the case.